src/Garn/GarnConfig.hs

```diff
@@ -82,7 +82,8 @@ data TargetConfig
 data ProjectTarget = ProjectTarget
   { description :: String,
     packages :: [String],
-    checks :: [String]
+    checks :: [String],
+    executable :: Bool
```

Comment:
```suggestion
    runnable :: Bool
```

Or maybe `isExecutable`? Or `hasDefaultExecutable`? I think `executable` is not ideal, since we do have a type called `Executable` and that might be confusing. I think I'd go with `runnable`.

(And then rename `isExecutable` in `OptParse` to `isRunnable`. I think `isExecutable` there is also kind of a confusing name.)